sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/TimeSeriesInsightsClientSample/InstancesSamples.cs

```diff
@@ -110,44 +133,58 @@ await foreach (TimeSeriesInstance tsiInstance in tsiInstances)
                 instanceId,
             };
 
-            Response<InstancesOperationResult[]> getInstancesByNameResult = await client.Instances.GetAsync(timeSeriesIds).ConfigureAwait(false);
+            Response<InstancesOperationResult[]> getByIdsResult = await client.Instances.GetAsync(timeSeriesIds).ConfigureAwait(false);
 
             /// The response of calling the API contains a list of instance or error objects corresponding by position to the array in the request.
             /// Instance object is set when operation is successful and error object is set when operation is unsuccessful.
-            InstancesOperationResult getInstanceByIdResult = getInstancesByNameResult.Value[0];
-            if (getInstanceByIdResult.Instance != null)
-            {
-                Console.WriteLine($"Retrieved Time Series Insights instance with Id '{getInstanceByIdResult.Instance.TimeSeriesId}' and name '{getInstanceByIdResult.Instance.Name}'.");
-            }
-            else if (getInstanceByIdResult.Error != null)
+            for (int i = 0; i < getByIdsResult.Value.Length; i++)
             {
-                Console.WriteLine($"Failed to retrieve a Time Series Insights instnace with Id '{getInstanceByIdResult.Error.Message}'.");
+                InstancesOperationResult currentOperationResult = getByIdsResult.Value[i];
+
+                if (currentOperationResult.Instance != null)
+                {
+                    Console.WriteLine($"Retrieved Time Series Insights instance with Id '{currentOperationResult.Instance.TimeSeriesId}' and name '{currentOperationResult.Instance.Name}'.");
+                }
+                else if (currentOperationResult.Error != null)
+                {
+                    Console.WriteLine($"Failed to retrieve a Time Series Insights instance with Id '{timeSeriesIds[i]}'. Error message: '{currentOperationResult.Error.Message}'.");
+                }
             }
 
             #endregion Snippet:TimeSeriesInsightsGetnstancesById
 
             // Clean up
             try
             {
-                #region Snippet:TimeSeriesInsightsSampleDeleteInstance
+                #region Snippet:TimeSeriesInsightsSampleDeleteInstanceById
+
+                var instancesToDelete = new List<TimeSeriesId>
```

Comment:
Would be interested to know if you think this approach is more readable vs creating a new list within DeleteAsync at line 168.

Reply:
I did it this way because I am referencing to instancesToDelete in a couple of times.